cpp/src/io/json/nested_json.hpp

```diff
@@ -259,6 +259,23 @@ tree_meta_t get_tree_representation(
   rmm::cuda_stream_view stream        = cudf::default_stream_value,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief Traverse the tree representation of the JSON input in records orient format and populate
+ * the output columns indices and row offsets within that column.
+ *
+ * @param d_input The JSON input
+ * @param d_tree A tree representation of the input JSON string as vectors of node type, parent
+ * index, level, begin index, and end index in the input JSON string
+ * @param stream The CUDA stream to which kernels are dispatched
+ * @param mr Optional, resource with which to allocate
+ * @return A tuple of the output column indices and the row offsets within each column for each node
+ */
+std::tuple<rmm::device_uvector<NodeIndexT>, rmm::device_uvector<size_type>>
+records_orient_tree_traversal(
+  device_span<SymbolT const> d_input,
+  tree_meta_t& d_tree,
+  rmm::cuda_stream_view stream        = cudf::default_stream_value,
```

Comment:
```suggestion
  rmm::cuda_stream_view stream,
```
Do we need to remove the default value? According to [the developer guide](https://github.com/rapidsai/cudf/blob/branch-22.10/cpp/doxygen/developer_guide/DEVELOPER_GUIDE.md#streams):
```
The internal detail API has the same parameters as the public API, plus a rmm::cuda_stream_view parameter at the end with no default value.
```
but I see streams with default values all over the place.

Reply:
Updated. Thanks.